src/scanner.c

```diff
@@ -187,6 +190,9 @@ prepare_scan(ScannerCtx *ctx)
 	ctx->internal.ended = false;
 	ctx->internal.registered_snapshot = false;
 
+	if (ctx->internal.scan_mcxt == NULL)
+		ctx->internal.scan_mcxt = CurrentMemoryContext;
+
```

Comment:
Isn't it a bug if this happens? Shouldn't the scan memory context always be set up?

Reply:
We can decide, but it is not setup in existing code that doesn't use the iterator interface. For instance, in many places we set up the Scanner context like so:

```
	ScannerCtx scanctx = {
		.table = catalog_get_table_id(catalog, CHUNK),
		.index = catalog_get_index(catalog, CHUNK, CHUNK_SCHEMA_NAME_INDEX),
		.nkeys = 1,
		.scankey = scankey,
		.tuple_found = chunk_rename_schema_name,
		.data = new_schema,
		.lockmode = RowExclusiveLock,
		.scandirection = ForwardScanDirection,
	};

```

The intention is to not have to touch the internal context (which is embedded in the `ScennerCtx` struct). We can change this if we want to, but I don't think it is necessary.